src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuckModule.java

```diff
@@ -1,52 +1,118 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import ucar.nc2.Attribute;
 import ucar.nc2.NetcdfFile;
 import ucar.nc2.NetcdfFileWriter;
+import ucar.nc2.Variable;
+import ucar.nc2.dataset.CoordinateAxis;
+import ucar.nc2.dataset.NetcdfDataset;
+import ucar.nc2.dataset.VariableDS;
+import ucar.nc2.dt.grid.GeoGrid;
+import ucar.nc2.dt.grid.GridDataset;
+
+import javax.servlet.ServletContext;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public class GoGoDuckModule {
     private static final Logger logger = LoggerFactory.getLogger(GoGoDuckModule.class);
 
-    // TODO Should not be hard coded
-    private static final String TIME_FIELD = "time";
-    private static final String URL_FIELD = "file_url";
+    @Autowired
+    ServletContext context;
+
+    private String profile;
+    private GoGoDuckSubsetParameters subset;
+    private IndexReader indexReader;
+    private GoGoDuckConfig goGoDuckConfig;
 
-    protected String profile = null;
-    protected GoGoDuckSubsetParameters subset = null;
-    protected UserLog userLog = null;
-    protected IndexReader indexReader = null;
+    private Boolean containsPackedVariable = null;
+    private List<String> ncksExtraParameters = null;
+    private NcksSubsetParameters ncksSubsetParameters = null;
+    private CoordinateAxis time = null, latitude = null, longitude = null;
 
-    public GoGoDuckModule() {}
+    public GoGoDuckModule() {
 
-    public void init(String profile, IndexReader indexReader, String subset, UserLog userLog) {
+    }
+
+    public GoGoDuckModule(String profile, IndexReader indexReader, String subset, GoGoDuckConfig goGoDuckConfig) {
         this.profile = profile;
         this.indexReader = indexReader;
         this.subset = new GoGoDuckSubsetParameters(subset);
-        this.userLog = userLog;
+        this.goGoDuckConfig = goGoDuckConfig;
     }
 
     public URIList getUriList() throws GoGoDuckException {
-        return indexReader.getUriList(profile, TIME_FIELD, URL_FIELD, subset);
+        return indexReader.getUriList(profile, goGoDuckConfig.getTimeField(), goGoDuckConfig.getFileUrlField(), subset);
+    }
+
+    public boolean hasPackedVariables() {
+        return containsPackedVariable;
     }
 
-    public void postProcess(File file) {
-        return;
+    public boolean isTimeUnlimited() {
+        return time.isUnlimited();
+    }
+
+    public void loadFileMetadata(File file) {
+        boolean containsPackedVariable = false;
+        GridDataset gridDs = null;
+
+        try {
+            setTimeLatLon(file.getAbsoluteFile().toString());
+            if (goGoDuckConfig.getUnpackNetcdf(profile)) {
+                // Checking the config (Needed for TPAC)
+                this.containsPackedVariable = true;
```

Comment:
This will result in us trying to unpack files that have no packed variables and is a bit confusing because the file does not necessarily actually contain any packed variables.  Perhaps rename to unpack - and only set if the file contains packed variables and we have configured gogoduck to unpack packed variables.